Jenkinsfile

```diff
@@ -470,355 +453,395 @@ def testGo(prefix, packagesToTest) {
-              withCredentials([
-                [$class: 'StringBinding', credentialsId: 'citogo-flake-webhook', variable : 'CITOGO_FLAKE_WEBHOOK'],
-                [$class: 'StringBinding', credentialsId: 'citogo-aws-secret-access-key', variable : 'CITOGO_AWS_SECRET_ACCESS_KEY'],
-                [$class: 'StringBinding', credentialsId: 'citogo-aws-access-key-id', variable : 'CITOGO_AWS_ACCESS_KEY_ID'],
-                [$class: 'StringBinding', credentialsId: 'citogo-master-fail-webhook', variable : 'CITOGO_MASTER_FAIL_WEBHOOK']
-              ]) {
-                println "Running tests for ${testSpec.dirPath}"
-                def t = getOverallTimeout(testSpec)
-                timeout(activity: true, time: t.time, unit: t.unit) {
-                  sh "citogo --flakes 3 --fails 3 --build-id ${env.BUILD_ID} --branch ${env.BRANCH_NAME} --prefix ${testSpec.dirPath} --s3bucket ci-fail-logs --build-url ${env.BUILD_URL} --no-compile --test-binary ./${testBinary} --timeout 150s ${testSpec.citogo_extra || ''}"
-                }
-              }
+          withCredentials([
+            [$class: 'StringBinding', credentialsId: 'citogo-flake-webhook', variable : 'CITOGO_FLAKE_WEBHOOK'],
+            [$class: 'StringBinding', credentialsId: 'citogo-aws-secret-access-key', variable : 'CITOGO_AWS_SECRET_ACCESS_KEY'],
+            [$class: 'StringBinding', credentialsId: 'citogo-aws-access-key-id', variable : 'CITOGO_AWS_ACCESS_KEY_ID'],
+            [$class: 'StringBinding', credentialsId: 'citogo-master-fail-webhook', variable : 'CITOGO_MASTER_FAIL_WEBHOOK']
+          ]) {
+            println "Running tests for ${testSpec.dirPath}"
+            def t = getOverallTimeout(testSpec)
+            timeout(activity: true, time: t.time, unit: t.unit) {
+              sh "citogo --flakes 3 --fails 3 --build-id ${env.BUILD_ID} --branch ${env.BRANCH_NAME} --prefix ${testSpec.dirPath} --s3bucket ci-fail-logs --build-url ${env.BUILD_URL} --no-compile --test-binary ./${testBinary} --timeout 150s ${testSpec.citogo_extra || ''}"
             }
           }
-          if (testSpec.name in specialTestFilter) {
-            specialTests["${prefix}${testSpec.name}"] = test
-          } else {
-            tests["${prefix}${testSpec.name}"] = test
-          }
-          println "Will run tests for ${testSpec.dirPath}"
-        } else {
-          println "Skipping tests for ${testSpec.dirPath} because no test binary was produced."
         }
       }
     }
-
-    // Schedule the tests
-    def parallelTests = []
-    def testBatch = [:]
-    tests.each { name, closure ->
-      if (testBatch.size() == 4) {
-        parallelTests << testBatch
-        testBatch = [:]
-      }
-      testBatch[name] = closure
+    if (testSpec.name in specialTestFilter) {
+      specialTests["${prefix}${testSpec.name}"] = test
+    } else {
+      tests["${prefix}${testSpec.name}"] = test
     }
-    if (testBatch.size() > 0) {
+  }
+  }
+
+  stage("Run Test Suite") {
```

Comment:
It doesn't really make sense to put a stage inside a closure...the scoping seems super weird to me.

Reply:
sorry, idk what a stage is. I just wanted logs for timing purposes

Reply:
i'll remove it

Reply:
that being said, it doesn't seem to be in a closure?

Reply:
There are variables outside this scope that get used inside this scope, hence closure.